components/html-block/html-block.js

```diff
@@ -193,13 +198,12 @@ class HtmlBlock extends RtlMixin(LitElement) {
 		this.noDeferredRendering = false;
 		this._hasSlottedContent = false;
 
-		const rendererContextAttributes = getRenderers().reduce((attrs, currentRenderer) => {
+		getRenderers().then(renderers => renderers.reduce((attrs, currentRenderer) => {
 			if (currentRenderer.contextAttributes) currentRenderer.contextAttributes.forEach(attr => attrs.push(attr));
 			return attrs;
-		}, []);
-
-		if (rendererContextAttributes.length === 0) return;
-		this._contextObserverController = new HtmlAttributeObserverController(this, ...rendererContextAttributes);
+		}, [])).then(rendererContextAttributes => {
+			this._contextObserverController = new HtmlAttributeObserverController(this, ...rendererContextAttributes);
+		});
```

Comment:
Should there be an extra `then` statement here that includes a call to `requestUpdate`? Or put another way, are we relying on `HtmlAttributeObserverController` calling `requestUpdate` due to the context attributes being updated, and is that good/safe? FWIW, I'm not sure calling `requestUpdate` in the constructor is safe either, even though it would be several micro-tasks later... a full task might be necessary to call it.

Reply:
Right, so I originally did do something like that, but was leery about it because (like you said) who knows if the component would be rendered by then. The controller will always request an update when the attributes change, including when they're first set, and it kind of has to work that way for the controller to function, so my inclination is that this change is safe. Whether it's _good_ is a different question of course. If there's concern there, I can look at forcing an update explicitly in the component.